code/modules/projectiles/projectile/special.dm

```diff
@@ -94,8 +94,7 @@
 			M.bodytemperature = temperature
 			if(temperature > 500)//emagged
 				M.adjust_fire_stacks(0.5)
-				M.on_fire = 1
-				M.update_icon = 1

```

Comment:
I thought it was a caching var, so the expensive human `update_icons` wouldn't be called more than it needed to be... I'll hunt down any remaining usages of this variable, then, and excise them.